frontend/src/components/NavBar.js

```diff
@@ -69,21 +70,15 @@ class NavBar extends React.Component<{
     }
 
     fetchName() {
-        $.ajax({
-            url: '/rest/user',
-            dataType: 'json',
-            cache: false,
-            type: 'GET',
-            success: (data) => {
+        fetchWrapper
+            .get('/rest/user')
```

Comment:
Interesting approach. I was thinking it might be good to create different functions for each route? So, for example, a function called `getUser` would then call `fetchWrapper.get('/rest/user')`.

Reply:
as far as I can tell from these changes, we only call the `/rest/user` endpoint three times. does this constitute a function? might be good to keep it as general as possible

Reply:
I agree with @adiprerepa , many api calls are only used once or twice. I was just focusing on removing $.ajax and replacing those with fetch.

Reply:
Hmm, but I don't know if there's much value in replacing $.ajax with fetchWrapper, then -- we're not changing the complexity of the code at all.

I was thinking it would be good to create functions for each endpoint, because some endpoints may also require additional processing of input / output that can then be factored out as well -- for example, see the code that calls `/rest/cli/command` (appears [3x in the codebase](https://github.com/codalab/codalab-worksheets/blob/43286e6206b20dc8ff8ee75d94286ed2bc6c1306/frontend/src/components/worksheets/WorksheetTerminal.js)) and the PUT `/contents/blob` endpoint that [appears 6x in the codebase](https://github.com/codalab/codalab-worksheets/search?l=JavaScript&q=%22contents%2Fblob%22). What do you think?

Reply:
I think it makes more sense to move all related ajax calls to one place since we are looking into using SWR and removing jquery, I am working on this and have pushed some new changes.